documents/patterns.md

```diff
@@ -19,16 +19,121 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-# OSGi pattern in Apache Celix
+# Apache Celix / OSGi  patterns 
+There are several design patterns used in Apache Celix and OSGi.  
+
+## Whiteboard Pattern
+The whiteboard pattern is a pattern where the act of registering a service is enough to participate in or extend 
+an existing functionality.  
+A typical aspect of the whiteboard pattern is that an application should 
+still resolve and startup even if the bundle or bundles that pick up the whiteboard pattern services are absent.
```

Comment:
From the whiteboard perspective they are indeed optional dependencies. 
But this part of the documentation aims to explain that if you provide a whiteboard service, it will not automatically be picked up. You also need to start the whiteboard service for something to happen.